### ISSUE TITLE
0 size text-shadow is throwing off color-contrast calculation

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2930]
The following code snippet has a contrast ratio of 4.84:1, but is reported by axe-core as having 3.49:1 because axe is misjudging the impact of the text-shadow:

```html
<style>
  a[href] {
    color: #0f833e;
    text-shadow: 0 0 0 #000;
    background-color: #fff;
  }
</style>
<a href="#">Hello World</a>
```

**Comment by WilcoFiers** [comment]
This is closely related to #2925 and #2815. Should probably address them in the same PR.

**Comment by straker** [comment]
This ticket should be closed/moved with https://github.com/dequelabs/axe-core/pull/3001 as the linked pr associated with this ticket was merged into a branch (and not develop).

**Comment by padmavemulapati** [comment]
Validated using latest axe-core develop branch code base, (dated 10th july, 2021) 
with the code snippet:
```
 <style>
    a[href] {
      color: #0f833e;
      text-shadow: 0 0 0 #000;
      background-color: #fff;
    }
  </style>
  <a href="#">Hello World</a>
```
Not seeing violations or best-practices for `color-contrast` rule on 0 size text-shadow elements.
![image](https://user-images.githubusercontent.com/47409579/125151486-2180cb00-e164-11eb-84aa-70af68c7a973.png)

and also pass check result showing color contrast ratio as expected.

![image](https://user-images.githubusercontent.com/47409579/125151546-98b65f00-e164-11eb-9371-2b693239a3f9.png)
